File: python/svix/models/environment_out.py
# this file is @generated
import typing as t
from datetime import datetime

from .common import BaseModel
from .connector_out import ConnectorOut
from .event_type_out import EventTypeOut


class EnvironmentOut(BaseModel):
    version: t.Optional[int] = None

    created_at: datetime

    event_types: t.List[EventTypeOut]

    settings: t.Optional[t.Dict[str, t.Any]]

    connectors: t.List[ConnectorOut]

File: kotlin/lib/src/main/kotlin/models/EnvironmentOut.kt
// This file is @generated
package com.svix.kotlin.models

import com.svix.kotlin.StringAnyMapSerializer
import kotlinx.datetime.Instant
import kotlinx.serialization.Serializable

@Serializable
data class EnvironmentOut(
    val createdAt: Instant,
    val eventTypes: List<EventTypeOut>,
    @Serializable(with = StringAnyMapSerializer::class) val settings: Map<String, Any>? = null,
    val transformationTemplates: List<ConnectorOut>,
    val version: Long? = null,
)
